velox/functions/remote/utils/RemoteFunctionServiceProvider.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/remote/utils/RemoteFunctionServiceProvider.h"
+
+#include "thrift/lib/cpp2/server/ThriftServer.h"
+#include "velox/functions/remote/server/RemoteFunctionService.h"
+
+namespace facebook::velox::functions {
+
+RemoteFunctionServiceParams
+RemoteFunctionServiceProviderForLocalThrift::getRemoteFunctionServiceParams() {
+  folly::call_once(initializeServiceFlag_, [&]() { initializeServer(); });
```

Comment:
won't this make the second caller to immediately return and fail since the server won't be up yet? Should probably still call the waitForRunning() or similar to ensure any subsequent thrift calls will success.

Reply:
I don't think this is needed. 'folly::call_once' guarantees that the second call does not return until the active call finishes, and since 'initializeServer()' already does 'waitForRunning()', server will already be running. 
folly::call_once is a drop-in replacement for std::call_once, behavior of std::call_once is described here - https://fburl.com/gn4bn9xk